### PR TITLE
doc: update release notes

### PR DESCRIPTION
**Pull request by darahayes**
### Description

Prepares the release notes for our upcoming refactor release!

##### Checklist

- [ ] `npm test` passes
- [ ] `npm run build` works
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

File: docs/ref-release-notes.md [hunk 1]
```diff
@@ -6,8 +6,170 @@ sidebar_label: Release notes
 
 # Next
 
+The `<version number>` release is a significant refactor of the core internals of Offix.
```
Comment (wtrocki) [inline docs/ref-release-notes.md]: ..of the Offix end user api and internals

File: docs/ref-release-notes.md [hunk 2]
```diff
@@ -6,8 +6,170 @@ sidebar_label: Release notes
...
+    mutation: <mutation object parsed by gql>,
+    optimisticResponse: <optimistic response object>,
+    variables: <mutation variables>
+  }
+}
+```
+
+`ApolloQueueEntryOperation` objects have two top level fields:
+
+* `qid` - Queue ID. This ID is randomly generated and mostly used by the OfflineQueue
```
Comment (craicoverflow) [inline docs/ref-release-notes.md]:
```suggestion
* `qid` - Queue ID. This ID is randomly generated and mostly used by the `OfflineQueue`.
```

File: docs/ref-release-notes.md [hunk 3]
```diff
@@ -6,8 +6,170 @@ sidebar_label: Release notes
...
+* `op` - The operation. In `offix-client` It's of type `MutationOptions`, the options object passed into client.offlineMutate with some extra metadata set by `offix-client`.
```
Comment (craicoverflow) [inline docs/ref-release-notes.md]:
```suggestion
* `op` - The operation. In `offix-client` It's of type `MutationOptions`, the options object passed into `client.offlineMutate` with some extra metadata set by `offix-client`.
```

File: docs/ref-release-notes.md [hunk 4]
```diff
@@ -6,8 +6,170 @@ sidebar_label: Release notes
...
+
+* Ensure users have no pending offline operations before administering the update.
+* Manually migrate the data using custom application code.
+* If it's not critical, users can re-enter the data or redo their offline operations.
+
 ## Features
 
+### New Documentation Website!
+
+Our documentation website has been rebuilt using Docusaurus. Special thanks to [LakshanKarunathilake](https://github.com/LakshanKarunathilake) for the complete overhaul.
```
Comment (craicoverflow) [inline docs/ref-release-notes.md]:
```suggestion
Our documentation website has been rebuilt using Docusaurus. Special thanks to [@LakshanKarunathilake](https://github.com/LakshanKarunathilake) for the complete overhaul.
```

File: docs/ref-release-notes.md [hunk 5]
```diff
@@ -6,8 +6,170 @@ sidebar_label: Release notes
...
+
+### OfflineQueue is directly accessible on the client
+
+`client.queue` is now directly accessible. This opens up the possibility for your application to directly see the operations in the queue. It also means you can manually call `client.queue.forwardOperations()` to execute all operations in the queue.
+
 ### Offline client enables wiping out cache using persistor interface
 
 `offlineClient.persitor.purge()` method will wipe entire persistence layer for Apollo cache
```
Comment (craicoverflow) [inline docs/ref-release-notes.md]:
```suggestion
`offlineClient.persitor.purge()` method will wipe entire persistence layer for Apollo cache.
```

**Pull request merged**